### PR TITLE
[AutoTVM] Use popenpool in local_executor

### PR DESCRIPTION
**Pull request by shingjan**
This PR intends to replace `multiprocessing.pool` with `popenpool` in `autotvm/local_executor`.
@junrushao1994 @vinx13 @tqchen

### CODE REVIEW

File: python/tvm/autotvm/measure/measure_methods.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def build(self, measure_inputs):
                 futures.append(ret)
 
             for future in futures:
-                res = future.get()
+                res = future.result()
```
[the remark this answers is not in the document]
Comment (shingjan) [reply python/tvm/autotvm/measure/measure_methods.py]: exception handling is updated

File: python/tvm/autotvm/measure/measure_methods.py [hunk 2]
```diff
@@ -114,53 +114,43 @@ def build(self, measure_inputs):
...
-                                    MeasureErrorNo.COMPILE_HOST,
-                                    res.time_cost,
-                                    time.time(),
-                                )
-                            )
-                else:
-                    # return BuildResult
-                    results.append(res)
+                    )
+                except InstantiationError as ex:
```
Comment (vinx13) [inline python/tvm/autotvm/measure/measure_methods.py:135]: these parts are handled in https://github.com/apache/tvm/blob/421dbf14f44c390bda56ba82e0d992b9ece14bf4/python/tvm/autotvm/measure/measure_methods.py#L516-L517, we can keep the original handling. We only need to handle `TimeoutError` and `ChildProcessError` here

Comment (shingjan) [reply python/tvm/autotvm/measure/measure_methods.py]: done

File: python/tvm/autotvm/measure/measure_methods.py [hunk 3]
```diff
@@ -114,53 +114,24 @@ def build(self, measure_inputs):
...
-                            )
-                        else:  # tvm error
-                            results.append(
-                                MeasureResult(
-                                    (res.error,),
-                                    MeasureErrorNo.COMPILE_HOST,
-                                    res.time_cost,
-                                    time.time(),
-                                )
-                            )
```
Comment (vinx13) [inline python/tvm/autotvm/measure/measure_methods.py:160]: These error handling is still needed (although the exception was handled inside the build_func, but the result to be appended is a bit different from that was originally returned). It should be added before https://github.com/apache/tvm/blob/14aa374973560ecddb14e667d6e449b1f847a698/python/tvm/autotvm/measure/measure_methods.py#L118

Comment (shingjan) [reply python/tvm/autotvm/measure/measure_methods.py]: done

**Review by vinx13 — APPROVED** [review]

**Comment by vinx13** [comment]
@shingjan can you also check if custom build func still works for autotvm?

**Review by vinx13 — CHANGES_REQUESTED** [review]
found some changes needed

**Comment by shingjan** [comment]
With a custom build function like the one below, this PR runs fine for `tune_conv2d_cuda`:

```
from tvm.contrib import tar


def custom_build(*args):
    return tar.tar(*args)


custom_build.output_format = "tar"

measure_option = autotvm.measure_option(
    builder=autotvm.LocalBuilder(build_func=custom_build),
    runner=autotvm.LocalRunner(repeat=3, min_repeat_ms=100, timeout=4),
)
```

**Review by vinx13 — APPROVED** [review]

**Pull request merged**